xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less

```diff
@@ -168,13 +177,21 @@ body.content {
       & .buttonwrapper,
       & #user-menu-col {
         & a {
-          text-decoration: none;
-          // Make sure we don't take over the default behaviour on hover with too much specificity.
-          &:hover, &:focus {
-            text-decoration: underline
-          }
+          .not-inline-link();
         }
       }
+
+      /* Remove the underlining from links with the btn class.
+       Those are already styled differently from their neighboring text. */
+      & a.btn {
+        .not-inline-link();
+      }
+
+      // Remove the underlining from links with an icon. In the vast majority of cases, these are already spottable.
+      & a:has( img[alt=""]), // With the silk iconTheme
```

Comment:
I wonder if this should really filter on the alt tag. I think it makes sense that in general all links that contain images aren't underline as otherwise the image is underlined which looks strange.

Reply:
I added this filter when thinking that we need to select icons. We ruled that icons do not have semantics so they should always have an empty alt. An other possibility would have been to select by `[data-xwiki-lightbox="false"]` but IMO this is worse (we don't want our anchor display to rely on the lightbox feature).

Reply:
Soooo...

I was testing it out and it happens that any semantically relevant image in content still hits this selector :)
Any of those images can be opened with  lightbox. Recently I added a hidden button under this image, with an icon inside it. This icon makes it so that this selector is hit no matter the state of the attached image...

Reply:
My argument is that the same that applies to icons also applies to all images. But maybe we don't all agree on that?

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/3077/commits/126d0bea1e51be688061d09407d7f0918899d6dd :+1: